test.ts

```diff
@@ -1,60 +1,87 @@
-type ModelParams = {
-    timestamp: string;
-    duration: number;
-    [key: string]: any;
-  };
+import { AggregationError } from './custom-errors';
+
+type Metric = string;

```

Comment:
Intentionally added this primitive type to highlight relations between `ModelParams`, `AggregationParams`, and `AggregatedMetrics`. With basic `string` it was not clear. I believe it made the code a bit more readable.